qa/rpc-tests/wallet_orchard.py

```diff
@@ -177,5 +177,54 @@ def run_test(self):
                 {'pools': {'orchard': {'valueZat': Decimal('100000000')}}, 'minimum_confirmations': 1},
                 self.nodes[3].z_getbalanceforaccount(acct3))
 
+        # Split the network again
+        self.split_network()
+
+        # Spend some of acct3's funds on the 2/3 side of the split
+        recipients = [{"address": ua2, "amount": Decimal('0.5')}]
+        myopid = self.nodes[3].z_sendmany(ua3, recipients, 1, 0)
+        rollback_tx = wait_and_assert_operationid_status(self.nodes[3], myopid)
+
+        self.sync_all()
+        self.nodes[2].generate(1)
+        self.sync_all()
+
+        assert_equal(
+                {'pools': {'orchard': {'valueZat': Decimal('950000000')}}, 'minimum_confirmations': 1},
+                self.nodes[2].z_getbalanceforaccount(acct2))
+
+        # Generate a majority chain on the 0/1 side of the split, then 
+        # re-join the network.
+        self.nodes[1].generate(10)
+        self.join_network()
+
+        # split 2/3's chain should have been rolled back, so their txn should have been
+        # un-mined and returned to the mempool
+        assert_equal(set([rollback_tx]), set(self.nodes[3].getrawmempool()))
+
+        # acct2's balance is back to not contain the Orchard->Orchard value
+        assert_equal(
+                {'pools': {'orchard': {'valueZat': Decimal('900000000')}}, 'minimum_confirmations': 1},
+                self.nodes[2].z_getbalanceforaccount(acct2))
+
+        # acct3's sole Orchard note is spent by a transaction in the mempool, so our
+        # confirmed balance is currently 0
+        assert_equal(
+                {'pools': {}, 'minimum_confirmations': 1},
+                self.nodes[3].z_getbalanceforaccount(acct3))
+
+        # Manually resend the transaction in node 3's mempool
+        self.nodes[2].resendwallettransactions()
```

Comment:
Adding this to the PR comments for posterity: originally this resent the transaction from node 3, but due to a combination of:
- the transaction being in node 2's wallet,
- the test framework using a linear topology 0-1-2-3,
- `join_network` connecting nodes 1 and 2 before the others when joining a split network,
- the transaction being rebroadcast intentionally from node 3,
the transaction ended up in node 2's mempool due to an internal rebroadcast without being broadcast immediately to node 1, and this then prevented the manual rebroadcast from node 3 from reaching node 1 (because node 2 ignores the broadcast as it already knows the transaction). This then caused the test to fail on the next `sync_all` which expects the mempools to sync. Manually rebroadcasting from node 2 solves this issue, ensuring that node 1 receives the tx.

This isn't a problem for the real network, because a) the topology is generally not linear, and b) rebroadcasts from node 2 _would_ happen after some timers run out (they just don't happen on the timescale of this RPC test).